packages/trpc/server/routers/viewer/availability/schedule/getAllSchedulesByUserId.handler.ts

```diff
@@ -0,0 +1,58 @@
+import { hasReadPermissionsForUserId } from "@calcom/lib/hasEditPermissionForUser";
+import { prisma } from "@calcom/prisma";
+
+import { TRPCError } from "@trpc/server";
+
+import type { TrpcSessionUser } from "../../../../trpc";
+import { getDefaultScheduleId } from "../util";
+import type { TGetAllByUserIdInputSchema } from "./getAllSchedulesByUserId.schema";
+
+type GetOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+  input: TGetAllByUserIdInputSchema;
+};
+
+export const getAllSchedulesByUserIdHandler = async ({ ctx, input }: GetOptions) => {
+  const { user } = ctx;
+
+  const isCurrentUserPartOfTeam = hasReadPermissionsForUserId({ memberId: input?.userId, userId: user.id });
+
+  const isCurrentUserOwner = input?.userId === user.id;
+
+  if (!isCurrentUserPartOfTeam && !isCurrentUserOwner) {
+    throw new TRPCError({
+      code: "UNAUTHORIZED",
+    });
+  }
+
+  const schedules = await prisma.schedule.findMany({
+    where: {
+      userId: input.userId,
+    },
+    select: {
+      id: true,
+      userId: true,
+      name: true,
+    },
+  });
+
+  if (!schedules) {
+    throw new TRPCError({
```

Comment:
Is the best thing to do here unauthorized? It's a situation that should never happen (but could happen) - but not sure that UNAUTHORIZED is the best approach.

Reply:
Thankyou! @emrysal for starting to review 🙏 

yeah may be is "NOT_FOUND" more apt here? Please let me know if this is ok.
Assuming that there will be at least one schedule for a given user, this may never happen, may be i added in case to handle the error if for some reason Prisma is not able to fetch the data.

Reply:
it's a error message for very rarely occurring scenario we can update.

Reply:
NOT_FOUND is good, also if you can add a logger to log this? We recently increased our log verbosity settings so it makes sense to log it using the logger.

Reply:
Thankyou! @emrysal  for confirming, 
updated error code to NOT_FOUND, also added log.

Reply:
@emrysal, can you please let know if you have any other comments or it is ok.